### PR TITLE
Filter prefixes

### PR DESCRIPTION
**Pull request by hrshdhgd**
- [x] Added a new function `filter_prefixes`: This filters the `msdf.df` to contain only those mappings with prefixes that are present in a list provided by the user.
 - [x] Add class method `remove_mappings()` to MSDF class. Addresses one checkbox in https://github.com/monarch-initiative/mondo-ingest/issues/49

### CODE REVIEW

**Review by matentzn — DISMISSED** [review]
I approve this, but if you really want to make me happy you can add a CLI method

```
remove(msdf1, msdf2)
  msdf1.remove_mappings(msdf2)
  write(msdf1)
```

**Comment by hrshdhgd** [comment]
Your wish is my command! 
```
sssom remove abcd.sssom.tsv --remove-map def.sssom.tsv -o abde.sssom.tsv` 
```
ashould work.

File: sssom/cli.py
Comment (matentzn) [inline sssom/cli.py:681]: This is great, we can later consider making this multivalued (not now). Thank you!

File: sssom/io.py
Comment (matentzn) [inline sssom/io.py:364]: this is new, I assume there was no test for the `else` case?

**Review by matentzn — APPROVED** [review]
Excellent, left a few comments but nothing that blocks merging. THANKS

File: sssom/io.py
Comment (hrshdhgd) [reply sssom/io.py:364]: This was missing and became apparent when only 1 value to a param was given for e.g. `--subject_id ABCD:1234` instead of multiple `--subject_id ABCD:1234 --subject_id DEFG:4567`. The missing `)` that's added in the code above threw an error. There was no test to check this.

**Pull request merged**
